### ISSUE TITLE
API's timeouts and /memories

### ISSUE THREAD

**Issue by beastoin** [BasedHardware/omi#1570]
<img width="892" alt="Image" src="https://github.com/user-attachments/assets/d71de6cf-e8af-4eb1-be68-6aa4011f2392" />

**2 problems:**

1. API doesn't have timeout. It could lead to the whole system outage -> APIs should have the timeout limits.
2. API /memories is slow, p99 in minutes -> should be in Ms.

**Comment by beastoin** [comment]
@nquang29 sry, not /messages. it's /memories 🫣

**Comment by krishpranav** [comment]
Hi, is this issue still open?

**Comment by beastoin** [comment]
yes @krishpranav . feel free to ask / colab with @nquang29 , looking fwd ☺️

**Comment by krishpranav** [comment]
how to contact @nquang29? is it on discord or github issue?. i am looking forward to fix the issue in this project.

**Comment by beastoin** [comment]
DM'ed in discord

**Comment by krishpranav** [comment]
I haven't received any. I've recently texted on the tasks-and-bounty channel.

**Comment by nquang29** [comment]
MR:
[1570](https://github.com/BasedHardware/omi/pull/1638)
